crates/noirc_evaluator/src/ssa_refactor/opt/inlining.rs

```diff
@@ -446,7 +454,14 @@ impl<'function> PerFunctionContext<'function> {
                 if self.inlining_main {
                     self.context.builder.terminate_with_return(return_values.clone());
                 }
-                let block_id = self.translate_block(block_id, block_queue);
+                // The decision of how to continue this block is handled by
+                // `handle_function_returns` once we know whether there are multiple returns to
+                // consider. Note that we identify the block containing the return terminator
+                // using `current_block` instead of `translate_block`. This is because the
+                // inlining of other calls within the this block can split it into multiple
+                // blocks. As such, the mapping provided by `translate_block` is considered stale
+                // and unusable for this annotation.
+                let block_id = self.context.builder.current_block();
```

Comment:
Can we amend this comment to
1. Make it shorter if possible
2. Clarify the wording a bit. Instead of "(the result of) `translate_block` is considered stale and unusable" we could mention translate_block returns the start of a set of inlined blocks, where we want the end/current block for the purposes of this return instruction.
3. Going along with 2, we should add documentation to translate_block to mention it returns the first inlined block of a set of inlined blocks if there were multiple.